### DISCUSSION TITLE
Directory structure best practices - import resolve

### DISCUSSION THREAD

**Discussion by clarencedesmarais** [pytest-dev/pytest#8082] [Q&A] [answered]
Hi, this is essentially a repost of a question i posted here : https://stackoverflow.com/questions/65039210/python-running-unit-and-integration-tests-with-standard-directory-structure

Given 

![image](https://user-images.githubusercontent.com/71276060/100471501-6742c480-30a8-11eb-8732-1ec4376c5c79.png)

How can i import `src > mypkg > app`  into `tests > unit > app_test.py`

If anybody could help out it would be great :)

**Comment by clarencedesmarais** [comment, marked as answer]
Follow up :  you can ignore this question, ive found an answer. Posted it in my so post.

**Comment by RonnyPfannschmidt** [comment]
the general suggestion is to use either editable installs or real installs to test against (tox can help with that)
